File: algorithm/acwing/LanQiaoBeiDayOne2023/src/Week1/Day2/AcWing100/main.go
package main
import (
    "fmt"
    "os"
    "bufio"
    "strconv"
    )

const N = 100010

var (
    out = bufio.NewWriter(os.Stdout)
    in = bufio.NewScanner(os.Stdin)
    bs = make([]byte, 20000 * 1024)

    a, b [N]int
    pos, neg, n int
    )

func max(i, j int) int {if i > j {return i}; return j}
func abs(i int) int {if i > 0 {return i}; return -i}
func scanInt() int {
    in.Scan()
    a, _ := strconv.Atoi(in.Text())
    return a
}

func main() {
    in.Buffer(bs, len(bs))
    n = scanInt()
    for i := 1; i <= n; i++ {
        a[i] = scanInt()
        b[i] = a[i] - a[i - 1]
    }

    for i := 2; i <= n; i++ {
        if b[i] > 0 {
            pos += b[i]
        } else {
            neg -= b[i]
        }
    }

    fmt.Fprintf(out, "%d\n", max(pos, neg))
    fmt.Fprintf(out, "%d\n",  abs(pos - neg) + 1)
    out.Flush()
}

File: algorithm/acwing/LanQiaoBeiDayOne2023/src/Week1/Day2/AcWing100/main.py
#author:袁佳林
def main():
    n = int(input())
    a = [0 for i in range(n+10)]
    for i in range(1,n+1):
        x = int(input())
        a[i] += x
        a[i+1] -= x

    pos, neg = 0, 0
    for i in range(2, n+1):
        if a[i] > 0:
            pos += a[i]
        else:
            neg -= a[i]

    min_cnt = min(pos, neg) + abs(pos - neg)
    res = abs(pos - neg) + 1

    print(min_cnt)
    print(res)
if __name__ == "__main__":
    main()
